erigon-lib/downloader/webseed.go

```diff
@@ -195,3 +204,96 @@ func (d *WebSeeds) readWebSeedsFile(webSeedProviderPath string) (snaptype.WebSee
 	d.logger.Debug("[snapshots.webseed] get from File provider", "urls", len(response), "file", fileName)
 	return response, nil
 }
+
+// downloadTorrentFilesFromProviders - if they are not exist on file-system
+func (d *WebSeeds) downloadTorrentFilesFromProviders(ctx context.Context, rootDir string) {
+	// TODO: need more tests, need handle more forward-compatibility and backward-compatibility case
+	//  - now, if add new type of .torrent files to S3 bucket - existing nodes will start downloading it. maybe need whitelist of file types
+	//  - maybe need download new files if --snap.stop=true
+	if !d.downloadTorrentFile {
+		return
+	}
+	if len(d.TorrentUrls()) == 0 {
+		return
+	}
+	var addedNew int
+	e, ctx := errgroup.WithContext(ctx)
+	urlsByName := d.TorrentUrls()
+	//TODO:
+	// - what to do if node already synced?
+	for name, tUrls := range urlsByName {
+		tPath := filepath.Join(rootDir, name)
+		if dir.FileExist(tPath) {
+			continue
+		}
+		addedNew++
+		if strings.HasSuffix(name, ".v.torrent") || strings.HasSuffix(name, ".ef.torrent") {
```

Comment:
Whitelist is better - then in future can add new types.

Reply:
what do you mean? this is your code here I just added the check that errors out if we have a torrent hash mismatching with our hardcoded ones

Reply:
this is all I added to your original code: https://github.com/ledgerwatch/erigon/blob/c976b01a11b6b33037456ad4ffd297ffc13fa073/erigon-lib/downloader/webseed.go#L295

Reply:
I mean: erigon 3 has also .kv.torrent files and maybe we will add other extensions in future.

Reply:
sure